Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2645,6 +2645,29 @@ static void adjustTextDirectionForCoalescedGeometries(const SimpleRange& range,
     }
 }
 
+static bool currentNodeRequiresToSeparateBoundary(const RenderObject* currentRenderer)
+{
+    return currentRenderer && currentRenderer->isOutOfFlowPositioned();
+}
+
+static bool isDescendantOfOutOfFlow(RenderObject* descendant, const RenderObject* stayWithin)
+{
+    for (RenderObject* current = descendant; current; current = current->parent()) {
```

Comment:
Nit - you could deploy smart pointers here too:

```suggestion
    for (CheckedPtr current = descendant; current; current = current->parent()) {
```